### PR TITLE
Con 4388

### PR DESCRIPTION
**Pull request by hkf-tech**

### CODE REVIEW

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Controllers/CohortController.cs [hunk 1]
```diff
@@ -361,21 +364,45 @@ public async Task<IActionResult> FileUploadReview(FileUploadReviewRequest reques
...
+            var viewModel = await _modelMapper.Map<BulkUploadAddDraftApprenticeshipsViewModel>(response);
+            viewModel.ProviderId = providerId;
+            return View(viewModel);
+        }
+
+        [HttpGet]
+        [Route("success", Name = RouteNames.SuccessSendToEmployer)]
+        public async Task<IActionResult> Success(long providerId)
+        {
+            var response = TempData.Get<BulkUploadAddAndApproveDraftApprenticeshipsResponse>(Constants.BulkUpload.ApprovedApprenticeshipResponse);
```
Comment (cofaulco) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Controllers/CohortController.cs:399]: Use `TempData.Peek()`?

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Cohort/BulkUploadAddDraftApprenticeshipsViewModelMapper.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+namespace SFA.DAS.ProviderCommitments.Web.Mappers.Cohort
+{
+    public class BulkUploadAddDraftApprenticeshipsViewModelMapper : IMapper<GetBulkUploadAddDraftApprenticeshipsResponse, BulkUploadAddDraftApprenticeshipsViewModel>
+    {
+        public Task<BulkUploadAddDraftApprenticeshipsViewModel> Map(GetBulkUploadAddDraftApprenticeshipsResponse source)
+        {   
+            var viewModel = new BulkUploadAddDraftApprenticeshipsViewModel
+            {                
+                BulkUploadDraftApprenticeshipsViewModel = (from result in source.BulkUploadAddDraftApprenticeshipsResponse
```
Comment (cofaulco) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Cohort/BulkUploadAddDraftApprenticeshipsViewModelMapper.cs:15]: Simplify to `.Select()` extension method syntax?

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Cohort/FileUploadReviewViewModelToBulkUploadAddDraftApprenticeshipsRequestMapper.cs [hunk 3]
```diff
@@ -23,44 +22,28 @@ public FileUploadReviewViewModelToBulkUploadAddDraftApprenticeshipsRequestMapper
 
         public async Task<BulkUploadAddDraftApprenticeshipsRequest> Map(FileUploadReviewViewModel source)
         {
-            var csVRecrods = await _cacheService.GetFromCache<List<Models.Cohort.CsvRecord>>(source.CacheRequestId.ToString());
+            var csVRecrods = await _cacheService.GetFromCache<List<CsvRecord>>(source.CacheRequestId.ToString());
```
Comment (cofaulco) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Cohort/FileUploadReviewViewModelToBulkUploadAddDraftApprenticeshipsRequestMapper.cs:25]: typo

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Cohort/ReviewApprenticeRequestToReviewApprenticeViewModelMapper.cs [hunk 4]
```diff
@@ -41,7 +41,8 @@ public async Task<ReviewApprenticeViewModel> Map(ReviewApprenticeRequest source)
             var csvRecords = await _cacheService.GetFromCache<List<CsvRecord>>(source.CacheRequestId.ToString());
             _logger.LogInformation("Total number of records from cache: " + csvRecords.Count);        
 
-            var groupedByCohort = csvRecords.Where(x => x.CohortRef == source.CohortRef );
+            var groupedByCohort = csvRecords.Where(x => x.CohortRef == source.CohortRef ||
```
Comment (cofaulco) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Cohort/ReviewApprenticeRequestToReviewApprenticeViewModelMapper.cs:44]: this var is called `groupByCohort but there is no grouping happening here only filtering?

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/appsettings.Development.json [hunk 5]
```diff
@@ -7,7 +7,7 @@
     }
   },
   "UseStubProviderRelationships": true,
-  "UseStubProviderAuth": true,
+  "UseStubProviderAuth": false,
```
Comment (cofaulco) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/appsettings.Development.json:10]: should we be checking in changes to these settings?

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/appsettings.json [hunk 6]
```diff
@@ -10,5 +10,5 @@
   },
   "UseStubProviderRelationships": false,
   "UseStubProviderAuth": false,
-  "UseLocalRegistry": false
+  "UseLocalRegistry": true
```
Comment (cofaulco) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/appsettings.json:13]: should we be checking in changes to these settings?

**Review by cofaulco — COMMENTED** [review]
a few comments

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/appsettings.json (hunk 6 above)
Comment (hkf-tech) [reply src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/appsettings.json:13]: Some one else checked in true few weeks ago, just reverting it back to the way it was before.

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/appsettings.Development.json (hunk 5 above)
Comment (hkf-tech) [reply src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/appsettings.Development.json:10]: some one chekced in true few weeks ago, reverting it to way it was before.

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/appsettings.json (hunk 6 above)
Comment (hkf-tech) [reply src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/appsettings.json:13]: This shouldn't go in, will make the change

**Review by cofaulco — APPROVED** [review]

**Pull request merged**